doc/spec/namespace.md

```diff
@@ -0,0 +1,137 @@
+# Namespace
+The namespace represents the full set of named repositories which can be
+referenced by Docker. A repository contains a set of content-addressable blobs
+and tags referencing those blobs. The namespace can be used to discover the
+location and trust information of a repository based on DNS, HTTP requests and
+other means. The namespace is global and owned through the domain at the root of
+the namespace path.
+
+A repository should always be referenced by its fully canonical value. If a
+client presents a shortened name to a user, that name should be fully
+canonicalized based on rules defined by the client before contacting a remote
+server. When mirroring repositories, the original name for a repository must be
+used. Changing a repository name is equivalent to copying or moving and will not
+be referenced by the original repository.
+
+## Format
+A name consists of two parts, a DNS host name plus a repository path. The host
+name follows the DNS rules without change. The total length of a name is 255
+characters however there is no specific limitation on DNS or path components.
```

Comment:
"The total length of a name is 255..." should be "The maximum length of a name is 255..."? This may also be worth clarifying with an example:

```
The canonical repository name `example.com/foo/bar` is 19 bytes long. The host
name `example.com` is 11 bytes long, leaving up to 244 bytes available for the
path component of the name, including a leading `/` path separator character.
```